internal/dag/status_test.go

```diff
@@ -29,147 +29,75 @@ import (
 )
 
 func TestDAGStatus(t *testing.T) {
-	secretRootsNS := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "ssl-cert",
-			Namespace: "roots",
-		},
-		Type: v1.SecretTypeTLS,
-		Data: secretdata(CERTIFICATE, RSA_PRIVATE_KEY),
-	}
-
-	secretContourNS := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "default-ssl-cert",
-			Namespace: "projectcontour",
-		},
-		Type: v1.SecretTypeTLS,
-		Data: secretRootsNS.Data,
-	}
-
-	fallbackSecret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "fallbacksecret",
-			Namespace: "roots",
-		},
-		Type: v1.SecretTypeTLS,
-		Data: secretdata(CERTIFICATE, RSA_PRIVATE_KEY),
-	}
-
-	serviceKuard := &v1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "kuard",
-			Namespace: secretRootsNS.Namespace,
-		},
-		Spec: v1.ServiceSpec{
-			Ports: []v1.ServicePort{{
-				Name:       "http",
-				Protocol:   "TCP",
-				Port:       8080,
-				TargetPort: intstr.FromInt(8080),
-			}},
-		},
-	}
 
-	serviceHome := &v1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "home",
-			Namespace: serviceKuard.Namespace,
-		},
-		Spec: v1.ServiceSpec{
-			Ports: []v1.ServicePort{{
-				Name:     "http",
-				Protocol: "TCP",
-				Port:     8080,
-			}},
-		},
+	type testcase struct {
+		objs                []interface{}
+		fallbackCertificate *types.NamespacedName
+		want                map[types.NamespacedName]Status
 	}
 
-	serviceFoo2 := &v1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "foo2",
-			Namespace: serviceKuard.Namespace,
-		},
-		Spec: v1.ServiceSpec{
-			Ports: []v1.ServicePort{{
-				Name:     "http",
-				Protocol: "TCP",
-				Port:     8080,
-			}},
-		},
-	}
+	tests := make(map[string]testcase)
```

Comment:
I don't think that this goes quite far enough - there is still 2500 lines of test cases and fixtures here. Could you apply the pattern of a local `run` function rather than putting the test cases in a table? It's a very similar result, except that it make it way easier to find the test case that failed.


```Go
run := func(desc string, tc testcase) {
        r.Run(desc, func(t * testing.T) {
        ...
        }
}

run("root proxy does not specify FQDN", testcase{
        ...
})
```

Reply:
Yeah, that's a great idea, I'll do that.

Reply:
Done.